test_users.py

```diff
@@ -0,0 +1,80 @@
+#Import unittest and os dependancy module
+import unittest
+import os
+
+# import JSON
+from flask import json
+import json
+
+from app import create_app
+
+
+class UsersTestsCase(unittest.TestCase):
+    #Set up methods for test cases
+    def setUp(self):
+    	#Initialize test variables
+        self.app = create_app("testing")
+        self.client = self.app.test_client
+        self.user_data = {
+    	   	'name':'John',
+    	   	'email': 'john@mail.com',
+    	   	'password': 'John2018',
+    	}
+        self.app_context = self.app.app_context()
+        self.app_context.push()
+
+
+    def test_user_registration(self):
+    	#Test api can register user
+    	result = self.client().post('/api/auth/register', data=self.user_data)
+    	self.assertEqual(result.status_code, 201)
+
+
+    def test_user_login(self):
+    	#test api allows user to login
+    	self.test_user_registration()
+    	result=self.client().post('/auth/login', data=dict(
+    		email='john@mail.com',
+    		password='John2018'
+    		), follow_redirects=True)
+    	self.assertIn(b'You are now logged in', result.data)
+
+
+    def test_user_logout(self):
+    	#test api allows user to logout
+    	#register and login first
+    	self.test_user_login
+    	#follow redirect to logout and message displayed
+    	res = self.client().get('/auth/logout', follow_redirects=True)
```

Comment:
Hey you don't need `follows_redirect` for  testing an API call

Reply:
Thank you for the pointer there. Much appreciated